### PR TITLE
Report write results over MQTT

### PR DESCRIPTION
**Pull request by ryanplusplus**

### CODE REVIEW

File: src/mqtt_client_adapter.cpp [hunk 1]
```diff
@@ -88,6 +88,41 @@ static void update_erd(i_mqtt_client_t* _self, tiny_erd_t erd, const void* _valu
...
+    erd_string = "0" + erd_string;
+  }
+
+  auto topic = String("geappliances/") + self->device_id + "/erd/0x" + erd_string + "/write_result";
+
+  if(success) {
+    self->client->publish(topic.c_str(), "success", true);
+  }
+  else {
+    switch(failure_reason) {
```
Comment (cbeeler) [inline src/mqtt_client_adapter.cpp:106]: Are we missing the busy case?

File: src/mqtt_client_adapter.cpp [hunk 2]
```diff
@@ -88,6 +88,41 @@ static void update_erd(i_mqtt_client_t* _self, tiny_erd_t erd, const void* _valu
...
+      case tiny_erd_client_write_failure_reason_not_supported:
+        self->client->publish(topic.c_str(), "not supported", true);
+        break;
+
+      case tiny_erd_client_write_failure_reason_incorrect_size:
+        self->client->publish(topic.c_str(), "incorrect size", true);
+        break;
+
+      default:
+        self->client->publish(topic.c_str(), "incorrect size", true);
```
Comment (cbeeler) [inline src/mqtt_client_adapter.cpp]: Should we use a different string here? `Unknown error`?

File: src/mqtt_client_adapter.cpp (hunk 1 above)
Comment (ryanplusplus) [reply src/mqtt_client_adapter.cpp:106]: No, that will manifest as "retries exhausted". We don't currently report that separately and, realistically, it should almost never happen.

File: src/mqtt_client_adapter.cpp (hunk 2 above)
Comment (ryanplusplus) [reply src/mqtt_client_adapter.cpp]: Whoops, this was supposed to be `"unknown"`

**Pull request merged**